[BUG] orchestrion fail to instrument if is not in the go.mod folder

<!--
If you have identified a specific bug in the orchestrion codebase, please describe it here. If the
issue requires debugging or investigation of your specific private setup, that is best accomplished via support. Please contact Datadog [support](http://docs.datadoghq.com/help/).
-->

**Version of orchestrion**
orchestrion v1.0.2
<!-- If you aren't using the [latest version](https://github.com/DataDog/orchestrion/releases) of orchestrion, try upgrading first to see if your issue has already been resolved -->

**Describe what happened:**

Orchestrion doesn't instrument if you aren't in the go.mod folder

```bash
orchestrion/_integration-tests/civisibility on  main [$?] via 🐹 v1.23.2 on ☁️  tony.redondo@datadoghq.com 
❯ ../../orchestrion go test -tags=integration -shuffle=on ./...                                                          
panic: Orchestrion is not enabled, please run this test with orchestrion

goroutine 1 [running]:
datadoghq.dev/orchestrion/_integration-tests/civisibility.TestMain(...)
	/Users/tony.redondo/repos/github/tonyredondo/orchestrion/_integration-tests/civisibility/main_test.go:30
main.main()
	_testmain.go:61 +0xa4
FAIL	datadoghq.dev/orchestrion/_integration-tests/civisibility	0.416s
FAIL

orchestrion/_integration-tests/civisibility on  main [$?] via 🐹 v1.23.2 on ☁️  tony.redondo@datadoghq.com took 3s 
❯ cd ..

orchestrion/_integration-tests on  main [$?] via 🐹 v1.23.2 on ☁️  tony.redondo@datadoghq.com 
❯ ../orchestrion go test -tags=integration -shuffle=on ./civisibility/...
ok  	datadoghq.dev/orchestrion/_integration-tests/civisibility	4.355s

```

**Describe what you expected:**

Test building and passing


**Steps to reproduce the issue:**
<!-- If you can include a minimal code sample to reproduce the issue it can significantly reduce resolution time -->

already in the description

**Additional environment details (Version of Go, Operating System, etc.):**

go version go1.23.2 darwin/arm64

--- a/internal/toolexec/aspect/oncompile.go
+++ b/internal/toolexec/aspect/oncompile.go
@@ -91,13 +91,12 @@ func (w Weaver) OnCompile(cmd *proxy.CompileCommand) (err error) {
 		err = writeLinkDeps(cmd, &linkDeps, orchestrionDir)
 	}()
 
-	var goModDir string
-	if goMod, err := goenv.GOMOD("."); err != nil {
-		return err
-	} else {
-		goModDir = filepath.Dir(goMod)
-		log.Tracef("Identified module directory: %s\n", goModDir)
+	goMod, err := goenv.GOMOD(".")
+	if err != nil {
+		return fmt.Errorf("go env GOMOD: %w", err)
 	}
+	goModDir := filepath.Dir(goMod)
+	log.Tracef("Identified module directory: %s\n", goModDir)
 
 	cfg, err := config.NewLoader(goModDir, false).Load()
 	if err != nil {

--- a/internal/toolexec/aspect/oncompile.go
+++ b/internal/toolexec/aspect/oncompile.go
@@ -13,6 +13,7 @@ import (
 	"slices"
 	"strings"
 
+	"github.com/DataDog/orchestrion/internal/goenv"
 	"github.com/DataDog/orchestrion/internal/injector"
 	"github.com/DataDog/orchestrion/internal/injector/aspect"
 	"github.com/DataDog/orchestrion/internal/injector/config"
@@ -90,7 +91,15 @@ func (w Weaver) OnCompile(cmd *proxy.CompileCommand) (err error) {
 		err = writeLinkDeps(cmd, &linkDeps, orchestrionDir)
 	}()
 
-	cfg, err := config.NewLoader(".", false).Load()
+	var goModDir string
+	if goMod, err := goenv.GOMOD("."); err != nil {
+		return err
+	} else {
+		goModDir = filepath.Dir(goMod)
+		log.Tracef("Identified module directory: %s\n", goModDir)
+	}
+
+	cfg, err := config.NewLoader(goModDir, false).Load()
 	if err != nil {
 		return fmt.Errorf("loading injector configuration: %w", err)
 	}

--- a/main_test.go
+++ b/main_test.go
@@ -23,12 +23,41 @@ import (
 	"golang.org/x/mod/semver"
 )
 
-type testCase interface {
+func TestBuildFromModuleSubdirectory(t *testing.T) {
+	run := runner{dir: t.TempDir()}
+
+	run.exec(t, "go", "mod", "init", "github.com/DataDog/orchestrion.testing")
+	run.exec(t, "go", "mod", "edit", "-replace", fmt.Sprintf("github.com/DataDog/orchestrion=%s", rootDir))
+	require.NoError(t, os.Mkdir(filepath.Join(run.dir, "cmd"), 0o755))
+	require.NoError(t, os.WriteFile(filepath.Join(run.dir, "cmd", "main.go"), []byte(`package main
+
+import (
+	"log"
+
+	"github.com/DataDog/orchestrion/runtime/built"
+)
+
+func main() {
+	if !built.WithOrchestrion {
+		log.Fatalln("Not built with orchestrion 🤨")
+	}
+}
+`), 0o644))
+	orchestrionBin := buildOrchestrion(t)
+	run.exec(t, orchestrionBin, "pin")
+
+	// Run the command from a working directory that is NOT the module root, so we can ensure the
+	// configuration is appopriately loaded from the module's root anyway.
+	runCmd := runner{dir: filepath.Join(run.dir, "cmd")}
+	runCmd.exec(t, orchestrionBin, "go", "run", ".")
+}
+
+type benchCase interface {
 	baseline(b *testing.B)
 	instrumented(b *testing.B)
 }
 
-var testCases = map[string]func(b *testing.B) testCase{
+var benchCases = map[string]func(b *testing.B) benchCase{
 	"DataDog:orchestrion": benchmarkOrchestrion,
 	// normal build
 	"traefik:traefik": benchmarkGithub("traefik", "traefik", "./...", false),
@@ -41,7 +70,7 @@ var testCases = map[string]func(b *testing.B) testCase{
 }
 
 func Benchmark(b *testing.B) {
-	for name, create := range testCases {
+	for name, create := range benchCases {
 		b.Run(fmt.Sprintf("repo=%s", name), func(b *testing.B) {
 			tc := create(b)
 			b.Run("variant=baseline", func(b *testing.B) {
@@ -65,8 +94,8 @@ type benchGithub struct {
 	harness
 }
 
-func benchmarkGithub(owner string, repo string, build string, testbuild bool) func(b *testing.B) testCase {
-	return func(b *testing.B) testCase {
+func benchmarkGithub(owner string, repo string, build string, testbuild bool) func(b *testing.B) benchCase {
+	return func(b *testing.B) benchCase {
 		b.Helper()
 
 		tc := &benchGithub{harness{build: build, testbuild: testbuild}}
@@ -89,12 +118,16 @@ type benchOrchestrion struct {
 	harness
 }
 
-func benchmarkOrchestrion(_ *testing.B) testCase {
-	return &benchOrchestrion{harness{dir: rootDir, build: ".", testbuild: false}}
+func benchmarkOrchestrion(_ *testing.B) benchCase {
+	return &benchOrchestrion{harness{runner: runner{dir: rootDir}, build: ".", testbuild: false}}
+}
+
+type runner struct {
+	dir string // The directory where commands are to be executed
 }
 
 type harness struct {
-	dir       string // The directory in which the source code of the package to be built is located.
+	runner
 	build     string // The package to be built as part of the test.
 	testbuild bool   // Whether the package to be built is a test package.
 }
@@ -143,15 +176,15 @@ func (h *harness) instrumented(b *testing.B) {
 	require.NoError(b, err, "build failed:\n%s", output)
 }
 
-func (h *harness) exec(b *testing.B, name string, args ...string) {
+func (r *runner) exec(tb testing.TB, name string, args ...string) {
 	cmd := exec.Command(name, args...)
-	cmd.Dir = h.dir
-	cmd.Env = append(os.Environ(), "GOCACHE="+b.TempDir())
+	cmd.Dir = r.dir
+	cmd.Env = append(os.Environ(), "GOCACHE="+tb.TempDir())
 	output := bytes.NewBuffer(make([]byte, 0, 4_096))
 	cmd.Stdout = output
 	cmd.Stderr = output
 
-	require.NoError(b, cmd.Run(), "command failed: %s\n%s", cmd, output)
+	require.NoError(tb, cmd.Run(), "command failed: %s\n%s", cmd, output)
 }
 
 func (*harness) findLatestGithubReleaseTag(b *testing.B, owner string, repo string) string {
@@ -227,14 +260,14 @@ var (
 	orchestrionBin     string
 )
 
-func buildOrchestrion(b *testing.B) string {
-	b.Helper()
+func buildOrchestrion(tb testing.TB) string {
+	tb.Helper()
 
 	orchestrionBinOnce.Do(func() {
 		orchestrionBin = filepath.Join(rootDir, "bin", "orchestrion.exe")
 
 		cmd := exec.Command("go", "build", fmt.Sprintf("-o=%s", orchestrionBin), rootDir)
-		require.NoError(b, cmd.Run())
+		require.NoError(tb, cmd.Run())
 	})
 
 	return orchestrionBin

